### PR TITLE
Milestone 4: Comments & likes

### PR DESCRIPTION
**Pull request by bolah2009**
### Milestone 4: Comments & likes

- Create comments and likes models with associations and implement all requested features for comments and likes.
- Users can Like Posts.
- Users can Comment on Posts.
- Posts should always display with the post content, author, comments, and likes.
- Add unit and integrations tests!

### CODE REVIEW

File: spec/features/likes_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  let(:other_user) { FactoryBot.create(:user) }
+
+  before do
+    visit root_path
+    fill_in 'user_email',    with: other_user.email
+    fill_in 'user_password', with: other_user.password
+    click_button 'Log in'
+  end
+
+  scenario "user can like other user posts" do
```
Comment (mcrd25) [inline spec/features/likes_spec.rb]: This test is failing.
Need to address before I can give you the 👍 
[REQUIRED]

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  before_action :authenticate_user!
+
+  def new; end
+
+  def create
+    @comment = current_user.comments.build(comment_params)
+    redirect_to posts_path if @comment.save
```
Comment (mcrd25) [inline app/controllers/comments_controller.rb]: What happens if a comment is not saved successfully? Maybe you can add code to address that scenario, and test for it first in `comments_controller_spec` [OPTIONAL]

Same applies for like

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  def destroy
+    @comment = Comment.find_by(id: params[:id])
```
Comment (mcrd25) [inline app/controllers/comments_controller.rb]: what happens if it fails to delete? Maybe you can address this and test for it as well.
[OPTIONAL]

Same applies for likes

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  before_action :authenticate_user!
+
+  def new; end
```
Comment (mcrd25) [inline app/controllers/comments_controller.rb]: Are you using this action? If not, you should remove! It does not seem like you are [OPTIONAL]

File: spec/features/likes_spec.rb (hunk 1 above)
Comment (bolah2009) [reply spec/features/likes_spec.rb]: ### Hi @mcrd25, :raising_hand_man: 

Thanks for the review, :bowing_man: :+1: 
The test is not failing on the last commit. :thinking: 
Can you send the test result to reproduce this issue?

See the test results below:

``` sh
$ rspec

CommentsController
  #create
    as an authenticated user
      comment on a post
    as a guest
      returns a 302 response
      redirects to the sign-in page
  #destroy
    as an authenticated user
      delete a comment
    as a guest
      returns a 302 response
      redirects to the sign-in page

LikesController
  #create
    as an authenticated user
1
4639
2
      like on a post
    as a guest
      returns a 302 response
      redirects to the sign-in page
  #destroy
    as an authenticated user
      delete a like
    as a guest
      returns a 302 response
      redirects to the sign-in page

PostsController
  #index
    as an authenticated user
      responds successfully
      returns a 200 response
    as a guest
      redirects to the sign-in page
      returns a 302 response
  #create
    as an authenticated user
      adds a post
    as a guest
      returns a 302 response
      redirects to the sign-in page

Comments
  user can write comments on posts
  user can delete comments
  user can't delete other user comments

Likes
  user can like and unlike posts

Posts
  user sign in and creates a new post

Users
  create a new user
  log in and user

Comment
  is invalid without a user
  is invalid without a post
  is invalid without a content

Like
  is invalid without a user
  is invalid without a post

Post
  is invalid without a content
  is invalid without a user

User
  is valid with a name, email, and password
  is invalid without a name
  is invalid without an email address
  is invalid with a duplicate email address

Finished in 1.71 seconds (files took 0.99723 seconds to load)
37 examples, 0 failures
```

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (bolah2009) [reply app/controllers/comments_controller.rb]: @Torres-ssf, see this :point_up:  :thinking:

**Comment by azdrenymeri** [comment]
### Hello @bolah2009  and @Torres-ssf :wave:  
### Good job on this milestone :+1: 
#### You can move on to the next milestone :1st_place_medal:

**Pull request merged**